### PR TITLE
CSV Export

### PR DESCRIPTION
**Pull request by jolavb**
Add function and button that creates a csv export containing 100 rows for each trail containing that trail's hydrant information.

### CODE REVIEW

File: src/components/ImportExport.js [hunk 1]
```diff
@@ -181,6 +183,47 @@ class ImportExport extends React.Component {
...
 
+  generateCSV = () => {
+    const { hydrants, trails } = this.props;
+    const hydrantsRows = [
+      ['Trail_Name', 'Hyd_Index', 'Hyd_ID', 'Hyd_State', 'Hyd_Hours',
+      'Hyd_Gun', 'Hyd_Gpm', 'Hyd_Notes', 'Hyd_Elevation',
+      'Hyd_Target_Gallons', 'Hyd_Total_Gallons', 'Hyd_Cfm', 'Hyd_Pressure_Zone']
+    ]
+    trails.keySeq().forEach((trailId) => {
+      const trailName = trails.get(trailId).get('name')
```
Comment (caacree) [inline src/components/ImportExport.js]: This needs to be the old format trail name for this db schema. 

`trailName = trails.getIn([trailId, 'features', 0].get('description').split(',')[0]`

**Comment by caacree** [comment]
Look at resolving that issue of keeping the old format of trail names, both for the csv and for the kml description fields, but other than that lgtm. 👍

**Comment by jolavb** [comment]
Alright, I think this last push fixes things. Original Trail Names are preserved in the feature on upload for use in the description fields on export. If a new trail is exported it uses Trail_Name instead. Also made hydrant indexes in the KML consistent with csv export.

File: src/components/ImportExport.js [hunk 2]
```diff
@@ -151,22 +152,36 @@ class ImportExport extends React.Component {
...
+
+    trails.valueSeq().forEach((v) => {
+      let trailName = v.get('name').split(' ').join('_')
+      // Iterate through Trail's Features
+      v.get('features').forEach((f) => {
+        if (f.get('originalTrailName')) {
+          trailName = f.get('originalTrailName')
+        }
+        const description = trailName
+        f.unset('features')
```
Comment (caacree) [inline src/components/ImportExport.js]: This isn't great. We shouldn't have things we don't want like this added to the trail when it's modified and we have to take off again here. Can you modify the TrailReducer so it's like HydrantReducer's `updateHydrantFeatures` and only puts the attributes we want on the feature?

File: src/components/ImportExport.js [hunk 3]
```diff
@@ -181,6 +196,47 @@ class ImportExport extends React.Component {
...
 
+  generateCSV = () => {
+    const { hydrants, trails } = this.props;
+    const hydrantsRows = [
+      ['Trail_Name', 'Hyd_Index', 'Hyd_ID', 'Hyd_State', 'Hyd_Hours',
+      'Hyd_Gun', 'Hyd_Gpm', 'Hyd_Notes', 'Hyd_Elevation',
+      'Hyd_Target_Gallons', 'Hyd_Total_Gallons', 'Hyd_Cfm', 'Hyd_Pressure_Zone']
+    ]
+    trails.keySeq().forEach((trailId) => {
+      const trailName = trails.getIn([trailId, 'features'])[0].get('originalTrailName')
```
Comment (caacree) [inline src/components/ImportExport.js]: It doesn't look like `generateCSV` has to go through `exportFile`, so new trails won't have an `originalTrailName` assigned yet unless you exported them already, or am I off?

File: src/redux/reducers/HydrantReducer.js [hunk 4]
```diff
@@ -39,6 +39,7 @@ export default (state = initialState, action) => {
       const { hydrants } = state;
       const hydrant = action.data;
       const newHydrants = hydrants.set(hydrant.id, hydrant);
+      console.log(hydrant.toJS())
```
Comment (caacree) [inline src/redux/reducers/HydrantReducer.js]: No thanks.

File: src/redux/reducers/TrailReducer.js [hunk 5]
```diff
@@ -56,6 +56,7 @@ export default (state = initialState, action) => {
     }
     case TRAIL_MODIFIED: {
       const { id, editedFields } = action.data;
+
```
Comment (caacree) [inline src/redux/reducers/TrailReducer.js]: Not a big deal, but it's nice to give a quick read-through of "files changed" in your PR's and see if things like this snuck in. All changes should be improvements and have a reason (whether through the feature itself or to fix a code style / linting issue, etc).

File: src/components/ImportExport.js (hunk 2 above)
Comment (jolavb) [reply src/components/ImportExport.js]: Updated in clean up branch.

File: src/components/ImportExport.js (hunk 3 above)
Comment (jolavb) [reply src/components/ImportExport.js]: Good catch. Fixed w/ commit  3b1095c

File: src/redux/reducers/HydrantReducer.js (hunk 4 above)
Comment (jolavb) [reply src/redux/reducers/HydrantReducer.js]: Gone. Commit 666e91d

File: src/redux/reducers/TrailReducer.js (hunk 5 above)
Comment (jolavb) [reply src/redux/reducers/TrailReducer.js]: Gotcha, makes sense. I'll try and be a bit better about that.

**Pull request merged**